adr/003-mediation-fees.md

```diff
@@ -0,0 +1,146 @@
+# Mediation Fees calculation and message format
+
+* Status: proposed
+* Deciders: Augusto, Dominik, Karl, Paul
+* Date: 2019-05-07
+
+## Context and Problem Statement
+
+Mediating nodes demand fees for their mediation services. These fees should not only incentivize running a Raiden node, but also help balancing the network. This can be done by reducing the mediation fees for payments which leave channels in a more balanced state than before the payment (possibly even to a negative fee).
+
+## Decision Drivers
+
+* Balancing the network should be incentivized
+* Raiden nodes will act selfishly
+* Mediation fees should be flexible enough to accommodate arbitrary business models for parties operation Raiden nodes.
+
+## Considered Options
+
+1. A fixed fee + a proportional fee (flat + prop * amount)
+2. A fixed fee + a proportional fee + an imbalance fee based on a fixed imbalance measure
+3. A fixed fee + a proportional fee + submitting points of a function describing the imbalance fee
+4. Submitting points of a single function (dependent on the payment amount) which includes all fees
+
+## Decision Outcome
+
+Chosen option: **Option 3**, but with some important additions (see below).
+
+Option 1 would be sufficient for the short very term, but does not incentivize rebalancing. Option 2 does not work because no objective balance measure could be found that is valid across all business models / user interests. Option 4 would work but require sending a changed fee schedule after every transfer or mediation. Since this information is broadcast, that's a non-negligible amount of traffic.
+
+### Positive Consequences
+
```

Comment:
I'd add that balance fees can also be disabled.

Reply:
Under "Message Format" I write

> The three mediation fee components are sent separately. Each component is optional and will be assumed to be zero (no cost) if not included in the message.

Do you expect a different statement, or do you just want this information in another place of the document?

Reply:
I would just state it as an advantage of the solution.

Reply:
I've added

> Fee components are optional. A node only using flat fees does not have to submit complicated functions.